app/assets/javascripts/beak/widgets/ractives/code-editor.coffee

```diff
@@ -5,7 +5,8 @@ window.RactiveModelCodeComponent = Ractive.extend({
   , isReadOnly:        undefined # Boolean
   , lastCompiledCode:  undefined # String
   , lastCompileFailed:     false # Boolean
-  , procedureNames:           {}
+  , procedureNames:           {} # HashMap
```

Comment:
The type should be `Object[T, U]`, but please fill in what the actual types of `T` (keys) and `U` (values) are.

Reply:
`procedureNames:   Object[String, Number]`  ?

Reply:
`Object[String, Number]`

Reply:
`procedureNames: Object[String, Number]` gives error.

Reply:
The type signature is a comment.  `procedureNames: {} # Object[String, Number]`